src/cluster.c

```diff
@@ -4332,69 +4360,39 @@ void clusterReplyMultiBulkSlots(client *c) {
      *               3) node ID
      *           ... continued until done
      */
-
     int num_masters = 0;
     void *slot_replylen = addReplyDeferredLen(c);
-
-    dictEntry *de;
-    dictIterator *di = dictGetSafeIterator(server.cluster->nodes);
-    while((de = dictNext(di)) != NULL) {
-        clusterNode *node = dictGetVal(de);
-        int j = 0, start = -1;
-        int i, nested_elements = 0;
-
-        /* Skip slaves (that are iterated when producing the output of their
-         * master) and  masters not serving any slot. */
-        if (!nodeIsMaster(node) || node->numslots == 0) continue;
-
-        for(i = 0; i < node->numslaves; i++) {
-            if (nodeFailed(node->slaves[i])) continue;
-            nested_elements++;
-        }
-
-        for (j = 0; j < CLUSTER_SLOTS; j++) {
-            int bit, i;
-
-            if ((bit = clusterNodeGetSlotBit(node,j)) != 0) {
-                if (start == -1) start = j;
+    int i = 0;
+    clusterNode *last_node = NULL;
+    int start_slot = 0;
+    int end_slot = 0;
+    int loop_range = CLUSTER_SLOTS + 1;
+    for (i = 0; i < loop_range; i++) {
+        if (NULL == last_node) {
+            if (CLUSTER_SLOTS != i && NULL != server.cluster->slots[i]) {
+                /* A new slots_range_start. */
+                last_node = server.cluster->slots[i];
+                start_slot = i;
+                end_slot = i;
```

Comment:
Less indentation is generally easier to follow. Then this top block is just responsible for finding the first node.

```suggestion
        if (NULL == last_node) {
            if (CLUSTER_SLOTS == i || NULL == server.cluster->slots[i])
                continue;
            /* A new slots_range_start. */
            last_node = server.cluster->slots[i];
            start_slot = i;
            end_slot = i;
        }
```